--- /usr/share/easy-rsa/easyrsa.orig
+++ /usr/share/easy-rsa/easyrsa
@@ -316,7 +316,7 @@
 prog_exit() {
 	ESTAT=0
 	[ -n "$1" ] && ESTAT=$1
-	(stty echo 2>/dev/null) || set -o echo
+	(stty echo 2>/dev/null) || { (set -o echo 2>/dev/null) && set -o echo; }
 	echo "" # just to get a clean line
 	exit "$ESTAT"
 } # => prog_exit()
@@ -489,9 +489,20 @@
 
 hide_read_pass()
 {
-	(stty -echo 2>/dev/null) || set +o echo
-	read -r "$@"
-	(stty echo 2>/dev/null) || set -o echo
+	# shellcheck disable=SC2039
+	if stty -echo 2>/dev/null; then
+		read -r "$@"
+		stty echo
+	elif (set +o echo 2>/dev/null); then
+		set +o echo
+		read -r "$@"
+		set -o echo
+	elif (echo | read -r -s 2>/dev/null) ; then
+		read -r -s "$@"
+	else
+		warn "Could not disable echo. Password will be shown on screen!"
+		read -r "$@"
+	fi
 } # => hide_read_pass()
 
 # build-ca backend:
